symphony-bdk-core/src/test/java/com/symphony/bdk/core/client/loadbalancing/LoadBalancingStrategyTest.java

```diff
@@ -0,0 +1,147 @@
+package com.symphony.bdk.core.client.loadbalancing;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.core.client.ApiClientFactory;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.core.config.model.BdkLoadBalancingConfig;
+import com.symphony.bdk.core.config.model.BdkLoadBalancingMode;
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+import com.symphony.bdk.core.config.model.BdkServerConfig;
+import com.symphony.bdk.core.test.MockApiClient;
+import com.symphony.bdk.gen.api.SignalsApi;
+import com.symphony.bdk.gen.api.model.AgentInfo;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiRuntimeException;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class LoadBalancingStrategyTest {
+
+  @Test
+  public void testNewInstanceRandomLB() {
+    LoadBalancingStrategy loadBalancingStrategy = getLoadBalancingStrategy(BdkLoadBalancingMode.RANDOM);
+    assertEquals(RandomLoadBalancingStrategy.class, loadBalancingStrategy.getClass());
+  }
+
+  @Test
+  public void testNewInstanceRoundRobinLB() {
+    LoadBalancingStrategy loadBalancingStrategy = getLoadBalancingStrategy(BdkLoadBalancingMode.ROUND_ROBIN);
+    assertEquals(RoundRobinLoadBalancingStrategy.class, loadBalancingStrategy.getClass());
+  }
+
+  @Test
+  public void testNewInstanceExternalLB() {
+    LoadBalancingStrategy loadBalancingStrategy =
+        getLoadBalancingStrategy(BdkLoadBalancingMode.EXTERNAL, Arrays.asList(""));
+    assertEquals(ExternalLoadBalancingStrategy.class, loadBalancingStrategy.getClass());
+  }
+
+  @Test
+  public void testRoundRobinLbStrategy() {
+    LoadBalancingStrategy loadBalancingStrategy = getLoadBalancingStrategy(BdkLoadBalancingMode.ROUND_ROBIN,
+        Arrays.asList("agent1", "agent2", "agent3"));
+
+    List<String> basePaths = Stream.generate(() -> loadBalancingStrategy.getNewBasePath()).limit(4)
+        .collect(Collectors.toList());
+
+    assertEquals(Arrays.asList("https://agent1:443", "https://agent2:443", "https://agent3:443", "https://agent1:443"),
+        basePaths);
+  }
+
+  @Test
+  public void testRandomLbStrategy() {
```

Comment:
what if we are unlucky (or very lucky) and the same random number is generated each time :D ?
if we want to make it testable we would extract the random number generator as a dependecy so we change inject a fake one for the test
however it is maybe overkill here so maybe I would just check reduce the number to be 1 instead of 200?